R/endless_horse.R

```diff
@@ -9,17 +9,58 @@
 #' geometrically after the first iteration in order to keep the horse
 #' on screen for a while, but it \emph{will} keep going forever.  Or
 #' until you hit escape/Ctrl-C depending on your platform).
+#' @param what_color (character) A 
+#' \href{https://github.com/r-lib/crayon}{\code{crayon}}-suported text color
+#' for \code{what}.
+#' @param horse_color (character) A 
+#' \href{https://github.com/r-lib/crayon}{\code{crayon}}-suported text color
+#' for your steed.
 #' @examples \dontrun{
 #' endless_horse()
 #' endless_horse(endless = FALSE)
 #' }
-endless_horse <- function(what="Hello world!", endless = TRUE, wait = 0.5) {
+endless_horse <- function(what="Hello world!", endless = TRUE, wait = 0.5, 
+                          what_color = NULL, horse_color = NULL) {
+  
+  if (!is.null(what_color) & !is.character(what_color)) {
```

Comment:
Could pull all this stuff out into a util function

Reply:
this stuff?

Reply:
Yeah all the checks since they're now used in `endless_horse` and `say`. It's not a ton of repetition so probably fine to leave it as is, imo